bindings/serialize.h

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SERIALIZE
+#define SERIALIZE
+
+#include <memory>
+#include <map>
+#include "v8-profiler.h"
+
+std::vector<char> serializeHeapProfile(std::unique_ptr<v8::AllocationProfile>
+  profile, int64_t intervalBytes, int64_t startTimeNanos);
+
+std::vector<char> serializeTimeProfile(std::unique_ptr<v8::CpuProfile>,
+  int64_t samplingIntervalMicros, int64_t startTimeNanos);
+
+class ProtoMessage {
+public:
+  virtual void encode(std::vector<char> &b) = 0;
+};
+
+class ValueType : public ProtoMessage {
+public:
+  int64_t typeX;
+  int64_t unitX;
+  ValueType(int64_t typeX=0, int64_t unitX=0);
+  virtual void encode(std::vector<char> &b);
+};
+
+class Label : public ProtoMessage {
+private:
+  int64_t keyX;
+  int64_t strX;
+  int64_t num;
+  int64_t unitX;
+public:
+  Label(int64_t keyX=0, int64_t strX=0, int64_t num=0, int64_t unitX=0);
+  virtual void encode(std::vector<char> &b);
+};
+
+class Mapping : public ProtoMessage {
+private:
+  uint64_t id;
+  uint64_t start;
+  uint64_t limit;
+  uint64_t offset;
+  uint64_t fileX;
+  uint64_t buildIDX;
+  bool hasFunctions;
+  bool hasFilenames;
+  bool hasLineNumbers;
+  bool hasInlineFrames;
+public:
+  Mapping(uint64_t id, uint64_t start, uint64_t limit, uint64_t offset,
+      uint64_t fileX, uint64_t buildIDX, bool hasFunctions, bool hasFilenames,
+      bool hasLineNumbers, bool hasInlineFrames);
+  virtual void encode(std::vector<char> &b);
+};
+
+class Line : public ProtoMessage {
+private:
+  uint64_t functionIDX;
+  int64_t line;
+public:
+  Line(uint64_t functionIDX, int64_t line);
+  virtual void encode(std::vector<char> &b);
+};
+
+class ProfFunction : public ProtoMessage {
```

Comment:
Prof -> Profile or just drop the prefix.

Reply:
I'd gotten compilation errors when the prefix wasn't present.
Will switch to Profile